lib/src/lexer/token.dart

```diff
@@ -43,6 +43,23 @@ enum TokenType {
   /// The comma token (`,`).
   comma,
 
+  /// /// The double literal.
+  ///
+  /// It follows the following grammar:
+  /// ```ebnf
+  /// <digit_separator> ::= "_"
+  /// <integer_literal> ::= <digit>+ ( <digit_separator> <digit>+ )*
```

Comment:
We should change the grammar to accommodate to the fact that we now allow multiple separators (like `1__000_000__000_000`).